mrmustard/physics/gaussian_integrals.py

```diff
@@ -149,9 +149,27 @@ def complex_gaussian_integral(
     return A_post, b_post, c_post
 
 
+def join_cs(
+    c1: ComplexTensor,
+    c2: ComplexTensor,
+):
+    r"""Joins two multivariate polynomial arrays into a single multivariate polynomial array."""
+    c1 = math.astensor(c1)
+    c2 = math.astensor(c2)
```

Comment:
I think you should be able to avoid these calls to `astensor` (avoids additional copies of the tensors)

Reply:
Easiest is probably the `astensor` call but it seems to me the "proper" solution would be to do the rewrite? This would avoid these calls, be consistent with the type hint and consistent with the bargmann

Reply:
Side question: do we expect users to be directly calling `physics.triplets`?

Reply:
I don't think so, but that is a question for Filippo.

Reply:
surely some users will do that, yes